### DISCUSSION TITLE
filemanager not working in gnome

### DISCUSSION THREAD

**Discussion by arun54321** [lxqt/lxqt#2188] [General]
I am not sure whether it's a bug or not. When I try to run file manager, it doesn't open.
![s](https://user-images.githubusercontent.com/42943340/153354526-21bd6ad1-1f41-43bb-8dd4-31795e90250a.jpg)

**Comment by stefonarch** [comment]
Did you test what the terminal says? Add it to` ~/.profile`

**Reply by arun54321** [reply]
```
QT_QPA_PLATFORM=wayland /usr/bin/pcmanfm-qt
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
QSocketNotifier: Can only be used with threads started with QThread
[1]    18758 segmentation fault (core dumped)  QT_QPA_PLATFORM=wayland /usr/bin/pcmanfm-qt
```

**Reply by tsujan** [reply]
Gnome ignores `~/.profile`. Use `~/.config/environment.d/SOME_FILE`. If it doesn't work, report the bug to Gnome.

EDIT: `SOME_FILE` should end with `.conf`, like `qt.conf`. See https://www.freedesktop.org/software/systemd/man/environment.d.html

**Reply by arun54321** [reply]
Is it the bug related to gnome? I can other x11 file manager like caja and thunar on gnome wayland without adding any environment variable.

**Reply by tsujan** [reply]
All of them are GTK-based. Gnome has no support for Qt, with which pcmanfm-qt is made. So, in a sense yes, it's Gnome's fault.

If you want to use Qt apps in Gnome, you'll need to set up a platform integration there. Your distro may have a doc about it.

**Reply by tsujan** [reply]
BTW, I had Gnome on my other laptop. pcmanfm-qt worked without needing any EV under its Wayland session. I also launched it with `pcmanfm-qt -platform wayland` and it started fine.
